e2e/tests/interchain_accounts/groups_test.go

```diff
@@ -0,0 +1,214 @@
+		GroupId: InitialGroupID, // always use the initial group id
+	})
+	s.Require().NoError(err)
+
+	return res.GroupPolicies[0].Address
+}
+
+func (s *InterchainAccountsGroupsTestSuite) TestInterchainAccountsGroupsIntegration() {
+	t := s.T()
+	ctx := context.TODO()
+
+	var (
+		groupPolicyAddr   string
+		interchainAccAddr string
+		err               error
+	)
+
+	// setup relayers and connection-0 between two chains
+	// channel-0 is a transfer channel but it will not be used in this test case
+	relayer, _ := s.SetupChainsRelayerAndChannel(ctx)
+	chainA, chainB := s.GetChains()
+
+	chainAWallet := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	chainAAddress := chainAWallet.Bech32Address(chainA.Config().Bech32Prefix)
+
+	chainBWallet := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
+	chainBAddress := chainBWallet.Bech32Address(chainB.Config().Bech32Prefix)
+
+	t.Run("create group with new threshold decision policy", func(t *testing.T) {
+		members := []grouptypes.MemberRequest{
+			{
+				Address: chainAAddress,
+				Weight:  DefaultGroupMemberWeight,
+			},
+		}
+
+		decisionPolicy := grouptypes.NewThresholdDecisionPolicy(DefaultGroupThreshold, DefaultVotingPeriod, DefaultMinExecutionPeriod)
+		msgCreateGroupWithPolicy, err := grouptypes.NewMsgCreateGroupWithPolicy(chainAAddress, members, DefaultMetadata, DefaultMetadata, true, decisionPolicy)
+		s.Require().NoError(err)
+
+		txResp, err := s.BroadcastMessages(ctx, chainA, chainAWallet, msgCreateGroupWithPolicy)
+		s.Require().NoError(err)
+		s.AssertValidTxResponse(txResp)
+
+	})
+
+	t.Run("submit proposal for MsgRegisterInterchainAccount", func(t *testing.T) {
+		groupPolicyAddr = s.QueryGroupPolicyAddress(ctx, chainA)
+		msgRegisterAccount := controllertypes.NewMsgRegisterInterchainAccount(ibctesting.FirstConnectionID, groupPolicyAddr, icatypes.NewDefaultMetadataString(ibctesting.FirstConnectionID, ibctesting.FirstConnectionID))
+
+		msgSubmitProposal, err := grouptypes.NewMsgSubmitProposal(groupPolicyAddr, []string{chainAAddress}, []sdk.Msg{msgRegisterAccount}, DefaultMetadata, grouptypes.Exec_EXEC_UNSPECIFIED)
+		s.Require().NoError(err)
+
+		txResp, err := s.BroadcastMessages(ctx, chainA, chainAWallet, msgSubmitProposal)
+		s.Require().NoError(err)
+		s.AssertValidTxResponse(txResp)
+	})
+
+	t.Run("vote and exec proposal", func(t *testing.T) {
+		msgVote := &grouptypes.MsgVote{
+			ProposalId: InitialProposalID,
+			Voter:      chainAAddress,
+			Option:     grouptypes.VOTE_OPTION_YES,
+			Exec:       grouptypes.Exec_EXEC_TRY,
+		}
+
+		txResp, err := s.BroadcastMessages(ctx, chainA, chainAWallet, msgVote)
+		s.Require().NoError(err)
+		s.AssertValidTxResponse(txResp)
+	})
+
+	t.Run("start relayer", func(t *testing.T) {
+		s.StartRelayer(relayer)
+	})
+
+	t.Run("verify interchain account registration success", func(t *testing.T) {
+		interchainAccAddr, err = s.QueryInterchainAccount(ctx, chainA, groupPolicyAddr, ibctesting.FirstConnectionID)
```

Comment:
I guess you could assert the addr is non empty

Reply:
Added as an extra sanity check but I don't think its reachable. Something would be wrong with the query then I think